bot/exts/recruitment/talentpool/_review.py

```diff
@@ -120,6 +130,77 @@ async def make_review(self, user_id: int) -> typing.Tuple[str, Optional[Emoji]]:
         review = "\n\n".join((opening, current_nominations, review_body, vote_request))
         return review, seen_emoji
 
+    async def archive_vote(self, message: PartialMessage, passed: bool) -> None:
+        """Archive this vote to #nomination-archive."""
+        message = await message.fetch()
+
+        # We consider that if a message has been sent less than 2 second before the one being archived
+        # it is part of the same nomination.
+        # For that we try to get messages sent in this timeframe until none is returned and NoMoreItems is raised.
+        messages = [message]
+        with contextlib.suppress(NoMoreItems):
+            async for new_message in message.channel.history(before=message.created_at):
+                if messages[-1].created_at - new_message.created_at > timedelta(seconds=2):
```

Comment:
This is a pretty strange metric to go by imo. From my testing, it usually works, but it can pretty easily break. If we have access to the message, why don't we check the content for the message opening (line 114). We can make part of the opening a constant to make this easier.

Reply:
The thing is that it should work with automatic messages too, that doesn't have a standard opening message. Maybe we could do that for bots only then? Not sure what the best course of action would be here.

Reply:
Hmm well, most votes will have pings. One option would be looking for messages with 2+ pings.

Reply:
That could work if we look for role pings, let's do that.

Reply:
Changed in 4ec0ec9